tests/components/homekit_controller/test_climate.py

```diff
@@ -147,6 +200,89 @@ async def test_climate_change_thermostat_temperature(hass, utcnow):
     assert helper.characteristics[TEMPERATURE_TARGET].value == 25
 
 
+async def test_climate_change_thermostat_temperature_range(hass, utcnow):
+    """Test that we can set separate heat and cool setpoints in heat_cool mode."""
+    helper = await setup_test_component(hass, create_thermostat_service)
+
+    await hass.services.async_call(
+        DOMAIN,
+        SERVICE_SET_HVAC_MODE,
+        {"entity_id": "climate.testdevice", "hvac_mode": HVAC_MODE_HEAT_COOL},
+        blocking=True,
+    )
+
+    await hass.services.async_call(
+        DOMAIN,
+        SERVICE_SET_TEMPERATURE,
+        {
+            "entity_id": "climate.testdevice",
+            "hvac_mode": HVAC_MODE_HEAT_COOL,
+            "target_temp_high": 25,
+            "target_temp_low": 20,
+        },
+        blocking=True,
+    )
+    assert helper.characteristics[TEMPERATURE_TARGET].value == 22.5
+    assert helper.characteristics[THERMOSTAT_TEMPERATURE_HEATING_THRESHOLD].value == 20
+    assert helper.characteristics[THERMOSTAT_TEMPERATURE_COOLING_THRESHOLD].value == 25
+
+
+async def test_climate_change_thermostat_temperature_range_iphone(hass, utcnow):
+    """Test that we can set all three set points at once (iPhone heat_cool mode support)."""
+    helper = await setup_test_component(hass, create_thermostat_service)
+
+    await hass.services.async_call(
+        DOMAIN,
+        SERVICE_SET_HVAC_MODE,
+        {"entity_id": "climate.testdevice", "hvac_mode": HVAC_MODE_HEAT_COOL},
+        blocking=True,
+    )
+
+    await hass.services.async_call(
+        DOMAIN,
+        SERVICE_SET_TEMPERATURE,
+        {
+            "entity_id": "climate.testdevice",
+            "hvac_mode": HVAC_MODE_HEAT_COOL,
+            "temperature": 22,
+            "target_temp_low": 20,
```

Comment:
Temperature and the range temp setpoints are meant to be exclusive in the service schema. I'm surprised that this works.